online-workflow/src/main/java/eu/itesla_project/online/tools/PrintOnlineWorkflowPostContingencyViolationsTool.java

```diff
@@ -113,116 +116,104 @@ public void run(CommandLine line) throws Exception {
+            } else if (line.hasOption("contingency")) {
+                String contingencyId = line.getOptionValue("contingency");
+                Map<Integer, List<LimitViolation>> contingencyViolations = onlinedb.getPostContingencyViolations(workflowId, contingencyId);
+                if (contingencyViolations != null && !contingencyViolations.keySet().isEmpty()) {
+                    try (TableFormatter formatter = createFormatter(formatterFactory, tableFormatterConfig, writer)) {
+                        Integer[] stateIds = contingencyViolations.keySet().toArray(new Integer[contingencyViolations.keySet().size()]);
+                        Arrays.sort(stateIds);
+                        for (Integer stateId : stateIds) {
+                            List<LimitViolation> violations = contingencyViolations.get(stateId);
+                            if (violations != null && !violations.isEmpty()) {
+                                printValues(formatter, stateId, contingencyId, violations, violationsFilter);
+                            }
+                        }
+                    }
+                } else
+                    System.out.println("\nNo post contingency violations for workflow " + workflowId + " and contingency " + contingencyId);
+            } else {
+                Map<Integer, Map<String, List<LimitViolation>>> wfViolations = onlinedb.getPostContingencyViolations(workflowId);
+                if (wfViolations != null && !wfViolations.keySet().isEmpty()) {
+                    try (TableFormatter formatter = createFormatter(formatterFactory, tableFormatterConfig, writer)) {
+                        Integer[] stateIds = wfViolations.keySet().toArray(new Integer[wfViolations.keySet().size()]);
+                        Arrays.sort(stateIds);
+                        for (Integer stateId : stateIds) {
+                            Map<String, List<LimitViolation>> stateViolations = wfViolations.get(stateId);
+                            if (stateViolations != null && !stateViolations.keySet().isEmpty()) {
+                                String[] contingencyIds = stateViolations.keySet().toArray(new String[stateViolations.keySet().size()]);
+                                Arrays.sort(contingencyIds);
+                                for (String contingencyId : contingencyIds) {
+                                    List<LimitViolation> violations = stateViolations.get(contingencyId);
+                                    if (violations != null && !violations.isEmpty()) {
+                                        printValues(formatter, stateId, contingencyId, violations, violationsFilter);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                } else
+                    System.out.println("\nNo post contingency violations for workflow " + workflowId);
+            }
```

Comment:
Found duplicated code in PrintOnlineWorkflowPostContingencyViolationsTool.java and PrintOnlineWorkflowPostContingencyLoadflowTool